src/libopensc/card-piv.c

```diff
@@ -5570,9 +5570,9 @@ static int piv_match_card_continued(sc_card_t *card)
                         apdu.resplen = sizeof(swissbit_version_buf);
                         apdu.le = apdu.resplen;
                         r2 = sc_transmit_apdu(card, &apdu); /* on error swissbit_version == 0 */
-                        if (apdu.resplen == 4) {
-                                priv->swissbit_version = (swissbit_version_buf[0] << 24) | (swissbit_version_buf[1] << 16) |
-                                    (swissbit_version_buf[2] << 8) | swissbit_version_buf[3];
+                        if (apdu.resplen >= 3) {
+                                priv->swissbit_version = (swissbit_version_buf[0] << 16) | (swissbit_version_buf[1] << 8) |
+                                                         swissbit_version_buf[2];
                                 sc_log(card->ctx, "Swissbit card->type=%d, r=0x%08x version=0x%08x", card->type, r, priv->swissbit_version);
                         }
```

Comment:
If the response is not long enough `swissbit_version` is zero right?

Should we log something if the response is too short?